src/cli/domain/init-template/index.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const path = require('path');
+const scaffold = require('./scaffold');
+const installTemplate = require('./install-template');
+const createComponentDir = require('./create-component-dir');
+const initPackage = require('./init-package');
+const utils = require('./utils');
+
+module.exports = function(options, callback) {
+  const {
+    componentName,
+    templateType,
+    logger = console,
+    cli = 'npm'
+  } = options;
+  const componentPath = path.join(process.cwd(), componentName);
+  const local = /^\.+\/|^\//.test(templateType);
+  const template = utils.getPackageName(templateType).replace('-compiler', '');
+  const compiler = `${template}-compiler`;
+
+  createComponentDir({ logger, componentName, componentPath });
+
+  try {
+    // If the required compiler is already installed locally no need to install it from NPM again
```

Comment:
I think I could probably just drop the the comments as it's pretty self understandable imho. 💯  on avoiding comments!